tests/client/test_client.py

```diff
@@ -22,6 +22,16 @@ def test_get(server):
     assert response.elapsed > timedelta(0)
 
 
+def test_get_invalid_url(server):
+    with httpx.Client() as client:
+        with pytest.raises(httpx.InvalidURL):
+            client.get("invalid://example.org")
+        with pytest.raises(httpx.InvalidURL):
+            client.get("://example.org")
+        with pytest.raises(httpx.InvalidURL):
+            client.get("http://")
```

Comment:
(Nit) Considering using a parametrized test style? I enjoy the documentation purpose of `id=...`, and that each case runs as its own test...

Then should we back-port this to `test_async_client.py` as well?

```suggestion
@pytest.mark.parametrize(
    "url",
    [
        pytest.param("invalid://example.org", id="scheme-not-http(s)"),
        pytest.param("://example.org", id="no-scheme"),
        pytest.param("http://", id="no-host"),
    ],
)
def test_get_invalid_url(server, url):
    with httpx.Client() as client:
        with pytest.raises(httpx.InvalidURL):
             client.get(url)
```